gmi_egads/gmi_egads.c

```diff
@@ -0,0 +1,907 @@
+/******************************************************************************
+
+  Copyright 2014 Scientific Computation Research Center, 
+      Rensselaer Polytechnic Institute. All rights reserved.
+  
+  This work is open source software, licensed under the terms of the
+  BSD license as described in the LICENSE file in the top-level directory.
+
+*******************************************************************************/
+#include <math.h>
+#include <string.h>
+#include <stdio.h>
+
+#include "egads.h"
+
+#include <gmi.h>
+#include "gmi_egads.h"
+#include "gmi_egads_config.h"
+
+/** \brief initialize a gmi_model with an EGADS body and number of regions */
+struct gmi_model* gmi_egads_init(struct egObject *body, int numRegions);
+
+/** \brief initialize the model adjacency table for 3D regions */
+void gmi_egads_init_adjacency(int ***adjacency);
+
+
+/// initialized by `gmi_egads_start`
+static ego eg_context;
+/// initialized by `gmi_egads_load`
+static ego eg_model;
+/// initialized by `gmi_egads_load`
+static ego eg_body;
+
+typedef struct egads_ent
+{
+  ego ego_ent;
+  int dim;
+  int tag;
+} egads_ent;
+
+/// global array of model entities
+/// egads_global_ents[0] - array of model verts
+/// egads_global_ents[1] - array of model edges
+/// egads_global_ents[2] - array of model faces
+/// egads_global_ents[3] - array of model regions
+static egads_ent *egads_global_ents[4];
+
+typedef struct egads_iter
+{
+   egads_ent *ents;
+   int nelem;
+   int idx;
+} egads_iter;
+
+/// adjacency table to be used for the "dummy" 3D elements that EGADS doesn't
+/// natively track
+static int **adjacency_table[6];
+
+/// read adjacency table from binary file
+static void read_adj_table(const char* filename,
+                           int *nregions)
+{
+  char *sup_filename;
+  sup_filename = EG_alloc(strlen(filename)+4+1);
```

Comment:
document why +4+1, also I don't think you free this

Reply:
Fixed in a7e41e3